VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionService.java

```diff
@@ -16,183 +16,159 @@
 
 
     @PATCH("/api/discussion/v1/threads/{thread_id}/")
-    DiscussionThread setThreadVoted(@Path("thread_id") String threadId,
-                                    @Body VoteBody voteBody)
-            throws HttpException;
+    Call<DiscussionThread> setThreadVoted(@Path("thread_id") String threadId,
+                                          @Body VoteBody voteBody);
 
 
     @PATCH("/api/discussion/v1/comments/{comment_id}/")
-    DiscussionComment setCommentVoted(@Path("comment_id") String commentId,
-                                      @Body VoteBody voteBody)
-            throws HttpException;
+    Call<DiscussionComment> setCommentVoted(@Path("comment_id") String commentId,
+                                            @Body VoteBody voteBody);
 
 
     @PATCH("/api/discussion/v1/threads/{thread_id}/")
-    DiscussionThread setThreadFollowed(@Path("thread_id") String threadId,
-                                       @Body FollowBody followBody)
-            throws HttpException;
+    Call<DiscussionThread> setThreadFollowed(@Path("thread_id") String threadId,
+                                             @Body FollowBody followBody);
 
 
     @PATCH("/api/discussion/v1/threads/{thread_id}/")
-    DiscussionThread setThreadRead(@Path("thread_id") String threadId,
-                                   @Body ReadBody readBody)
-            throws HttpException;
+    Call<DiscussionThread> setThreadRead(@Path("thread_id") String threadId,
+                                         @Body ReadBody readBody);
 
 
     @POST("/api/discussion/v1/threads/")
-    DiscussionThread createThread(@Body ThreadBody threadBody)
-            throws HttpException;
+    Call<DiscussionThread> createThread(@Body ThreadBody threadBody);
 
 
     @FormUrlEncoded
     @POST("/api/discussion/v1/comments/")
-    DiscussionComment createComment(@Field("thread_id") String threadId,
-                                    @Field("raw_body") String rawBody,
-                                    @Field("parent_id") String parentId)
-            throws HttpException;
+    Call<DiscussionComment> createComment(@Field("thread_id") String threadId,
+                                          @Field("raw_body") String rawBody,
+                                          @Field("parent_id") String parentId);
+
+    final class FlagBody {
+        private boolean abuseFlagged;
+
+        public FlagBody(boolean abuseFlagged) {
+            this.abuseFlagged = abuseFlagged;
+        }
+    }
+
+    final class VoteBody {
+        private boolean voted;
+
+        public VoteBody(boolean voted) {
+            this.voted = voted;
+        }
+    }
+
+    final class FollowBody {
+        private boolean following;
+
+        public FollowBody(boolean following) {
+            this.following = following;
+        }
+    }
+
+    final class ReadBody {
+        private boolean read;
+
+        public ReadBody(boolean read) {
+            this.read = read;
+        }
+    }
```

Comment:
I just copied them over from the top-level to inner classes, since top-level classes with names different from their source files can't be declared as public; and they needed to be declared public since they were now going to be instantiated from any client code (from different packages), instead of just the wrapper *Service class (in the same package).